qualtran/_infra/controlled.py

```diff
@@ -458,9 +458,12 @@ def _unitary_(self):
             # subbloq is a cirq gate, use the cirq-style API to derive a unitary.
             import cirq
 
-            return cirq.unitary(
-                cirq.ControlledGate(self.subbloq, control_values=self.ctrl_spec.to_cirq_cv())
-            )
+            try:
```

Comment:
Why are we just passing? This is because the subbloq might not be unitary right? Maybe we should at least print a warning to indicate the user is doing something unsupported etc.

Reply:
Re-reading, if a warning is too annoying (although I guess we could leverage logging here) at least add a comment to the source code

Reply:
The ideal would be to always use `tensor_contract`, but at the moment, it's about 5-10x slower than `cirq.unitary` (e.g. #1336). So my interim fix was to try cirq, and use tensor if it failed.

I'll add this as a comment in the source code